source/MRMesh/MRMeshComponents.h

```diff
@@ -49,12 +49,27 @@ enum FaceIncidence
 
 /// given prepared union-find structure returns the union of connected components, each having at least given area
 [[nodiscard]] MRMESH_API FaceBitSet getLargeByAreaComponents( const MeshPart& meshPart, UnionFind<FaceId> & unionFind, float minArea,
-    UndirectedEdgeBitSet * bdEdgesBetweenLargeComps = nullptr );
+    UndirectedEdgeBitSet * outBdEdgesBetweenLargeComps = nullptr );
+
+struct LargeByAreaNumComponentsSettings
+{
+    /// return at most given number of largest by area connected components
+    int maxLargeComponents = 2;
+
+    /// do not consider a component large if its area is below this value
+    float minArea = 0;
+
+    /// optional predicate of boundaries between components
+    UndirectedEdgePredicate isCompBd;
+};
+
+/// returns requested number of largest by area connected components in descending by area order
+[[nodiscard]] MRMESH_API std::vector<FaceBitSet> getLargeByAreaNumComponents( const MeshPart& meshPart, const LargeByAreaNumComponentsSettings & settings );
```

Comment:
name is quite confusing, it seems to return number of large components and not components them-selfs